### ISSUE TITLE
Support ovs_multitenant plugin by allowing deployment to the default project

### ISSUE THREAD

**Issue by mwringe** [hawkular/hawkular-openshift-agent#123]
With the ovs_multitenant plugin, projects are more network isolated from each other and this means that the agent in the openshift-infra project cannot access the ip address for a pod in another project.

If we were able to run the agent in to the 'default' project, then it would work since the default project has an VNID of 0 which allow it to have network access across projects (this is where the registry and other components are deployed and is why they can still continue to work across projects).

The problem is that the agent relies on a secret from the 'openshift-infra' project, and secrets are tied to projects and cannot be used across projects.

However, the agent SA has additional permission to access secrets across projects. While this means we cannot specify the secrets to be used in the templates or object definitions, the agent can access this information programmatically from the OpenShift API.

An option to support this usecase would be to allow an option to gather the username/password directly via the REST API. The agent would then startup and wait for the secret to become available before continuing.

A readiness probe could also be used here to keep the agent in the 'not yet ready' state until this secret is available.

**Comment by mwringe** [comment]
@jmazzitelli any thoughts on this?

**Comment by ljuaneda** [comment]
Another solution is to turn the openshift-infra project global : 
https://docs.openshift.org/latest/admin_guide/managing_pods.html#making-project-networks-global

**Comment by mwringe** [comment]
The problem is that we might have other components running in openshift-infra which should not necessarily be exposed across the cluster. But yes, that looks like it would be another option.

**Comment by ljuaneda** [comment]
it doesn't work when deploying the agent in the default namespace :
https://github.com/openshift/origin-metrics#deploying-the-hawkular-openshift-agent

```
Unable to mount volumes for pod "hawkular-openshift-agent-1ap2y_default(5b7bde7e-faa7-11e6-a46f-001a64760e80)": timeout expired waiting for volumes to attach/mount for pod "hawkular-openshift-agent-1ap2y"/"default". list of unattached/unmounted volumes=[hawkular-metrics-certificate]
Error syncing pod, skipping: timeout expired waiting for volumes to attach/mount for pod "hawkular-openshift-agent-1ap2y"/"default". list of unattached/unmounted volumes=[hawkular-metrics-certificate]
MountVolume.SetUp failed for volume "kubernetes.io/secret/5b7bde7e-faa7-11e6-a46f-001a64760e80-hawkular-metrics-certificate" (spec.Name: "hawkular-metrics-certificate") pod "5b7bde7e-faa7-11e6-a46f-001a64760e80" (UID: "5b7bde7e-faa7-11e6-a46f-001a64760e80") with: secrets "hawkular-metrics-certificate" not found

```

**Comment by jmazzitelli** [comment]
> secrets "hawkular-metrics-certificate" not found

This sounds to me like Origin-Metrics isn't even installed. That is a requirement. Can you confirm that Origin-Metrics is deployed successfully?

**Comment by ljuaneda** [comment]
OpenShift-Metrics is deployed in openshift-infra.
But it seems I'm not running the latest version of the hawkular agent
image.


Le ven. 24 févr. 2017 à 19:29, John Mazzitelli <notifications@github.com> a
écrit :

**Comment by jmazzitelli** [comment]
> But it seems I'm not running the latest version of the hawkular agent image.

OK, definitely run the latest 1.2.3.Final - anything before won't run in any namespace other than "openshift-infra". To deploy in the "default" namespace (which is what we are recommending now, especially to fix this issue) you need the latest release.

**Comment by ljuaneda** [comment]
It's much better with the latest img version, still there is a pb with the hawkular certificate :

`
I0227 09:54:03.386437       1 prometheus_metrics_collector.go:97] DEBUG: Told to collect all Prometheus metrics from [http://10.131.2.32:8080/metrics]
W0227 09:54:03.425781       1 metrics_storage.go:163] Failed to store metrics. err=Post https://hawkular-metrics.openshift-infra.svc.cluster.local/hawkular/metrics/counters/raw: x509: certificate is valid for hawkular-metrics, metrics.osev3-qt9-infra.hpmetier.sf.intra.laposte.fr, not hawkular-metrics.openshift-infra.svc.cluster.local
`

I'm running openshift-metrics on OCP 3.4

**Comment by jmazzitelli** [comment]
@ljuaneda that's because the fix to this issue (#123 - support for ovs_multitenant) requires two fixes - one for the agent (which you now have) and one for origin-metrics (which it seems you do not have). Without both patches, it will not work.

See these PRs: (1) https://github.com/openshift/origin-metrics/pull/302  (2) https://github.com/hawkular/hawkular-openshift-agent/pull/124

And the BZ that started this all: https://bugzilla.redhat.com/show_bug.cgi?id=1421060

It has to do with the new way OpenShift Origin-Metrics generates its certificate - related to this commit which fixes the issue: https://github.com/openshift/origin-metrics/commit/17fdf950dc09061ecabad74b610b7b87b59eedad


In other words, in order to be able to use ovs_multitenant you must have that new code in agent and the new fix to origin-metrics - otherwise multitenant will not work. You cannot run the agent in the default namespace without both the latest agent and the latest origin-metrics that includes the new certificate.

Now that you have the latest agent, you should upgrade to the latest version of Origin-Metrics that includes that PR fix - that is the only way this issue #123 can be fixed.

@mwringe if there are any more details you want to add or correct anything above, please feel free.

**Comment by ljuaneda** [comment]
I saw the isssue [openshift/origin-metrics@17fdf95](https://github.com/openshift/origin-metrics/commit/17fdf950dc09061ecabad74b610b7b87b59eedad), but we'll wait it is released in the OCP channel.

I've figured how to make it work anyway, setting the hawkular-service url in the agent's configmap to the the route, and I've changed the route to passthrough.

**Comment by mwringe** [comment]
Closing this issue as it has been fixed in the latest origin-metrics and hosa
